raiden/blockchain/state.py

```diff
@@ -95,41 +93,23 @@ def get_contractreceivechannelsettled_data_from_event(
     if not channel_state:
         return None
 
-    # Recover the locksroot from the blockchain to fix data races. Check
-    # get_onchain_locksroots for details.
-    try:
-        # First try to query the unblinded state. This way the
-        # ContractReceiveChannelSettled's locksroots will  match the values
-        # provided during settle.
-        our_locksroot, partner_locksroot = get_onchain_locksroots(
-            chain=chain_service,
-            canonical_identifier=channel_state.canonical_identifier,
-            participant1=channel_state.our_state.address,
-            participant2=channel_state.partner_state.address,
-            block_identifier=block_hash,
-        )
-    except ValueError:
-        # State pruning handling. The block which generate the
-        # ChannelSettled event may have been pruned, because of this the
-        # RPC call raised ValueError.
-        #
-        # The solution is to query the channel's state from the latest
-        # *confirmed* block, this /may/ create a ContractReceiveChannelSettled
-        # with the wrong locksroot (i.e. not the locksroot used during the call
-        # to settle). However this is fine, because at this point the channel
-        # is settled, it is known that the locksroot can not be reverted
-        # without an unlock, and because the unlocks are fair it doesn't matter
-        # who called it, only if there are tokens locked in the settled
-        # channel.
-        our_locksroot, partner_locksroot = get_onchain_locksroots(
-            chain=chain_service,
-            canonical_identifier=channel_state.canonical_identifier,
-            participant1=channel_state.our_state.address,
-            participant2=channel_state.partner_state.address,
-            block_identifier=latest_confirmed_block,
+    order = order_locksroot(
```

Comment:
Why is the variable called `order`?  Isn't this ordered something?

Reply:
Humm, I didn't think of a better name. `order_locksroot` is just a routine that finds which of the locksroot was received by the current node, and which was sent, the variable `order` just has a struct which encodes this knowledge.